src/transports/http.ts

```diff
@@ -0,0 +1,259 @@
+      cacheMetrics.cacheWarmerCount.labels({ isBatched: 'true' }).inc()
+      this.WARMER_ACTIVE = true
+    }
+
+    logger.trace(`Have ${entries.length} entries in batch, preparing requests...`)
+    const rawRequests = this.config.prepareRequests(entries, context.adapterConfig)
+    const requests = Array.isArray(rawRequests) ? rawRequests : [rawRequests]
+
+    // We're awaiting these promises because although we have request coalescing, new entries
+    // could be added to the subscription set if not blocking this operation, so the next time the
+    // background execute is triggered if the request is for a fully batched endpoint, we could end up
+    // with the full combination of possible params within the request queue
+    logger.trace(`Sending ${requests.length} requests...`)
+    const start = Date.now()
+    await Promise.all(requests.map((r) => this.handleRequest(r, context.adapterConfig)))
+    const duration = Date.now() - start
+    logger.trace(`All requests in the background execute were completed`)
+
+    // These logs will surface warnings that operators should take action on, in case the execution of all
+    // requests is taking too long so that entries could have expired within this timeframe
+    if (duration > context.adapterConfig.WARMUP_SUBSCRIPTION_TTL) {
+      logger.warn(
+        `Background execution of all HTTP requests in a batch took ${duration},\
+         which is longer than the subscription TTL (${context.adapterConfig.WARMUP_SUBSCRIPTION_TTL}).\
+         This might be due to insufficient speed on the selected API tier, please check metrics and logs to confirm and consider moving to a faster tier.`,
+      )
+    }
+    if (duration > context.adapterConfig.CACHE_MAX_AGE) {
+      logger.warn(
+        `Background execution of all HTTP requests in a batch took ${duration},\
+         which is longer than the max cache age (${context.adapterConfig.CACHE_MAX_AGE}).\
+         This might be due to insufficient speed on the selected API tier, please check metrics and logs to confirm and consider moving to a faster tier.`,
+      )
+    }
+
+    return
+  }
+
+  private async handleRequest(
+    requestConfig: ProviderRequestConfig<T>,
+    adapterConfig: AdapterConfig<T['CustomSettings']>,
+  ): Promise<void> {
+    const results = await this.makeRequest(requestConfig, adapterConfig)
+
+    if (!results.length) {
+      logger.trace('Got no results from the request.')
+      return
+    }
+
+    logger.debug('Setting adapter responses in cache')
+    await this.responseCache.write(results)
+  }
+
+  private async makeRequest(
+    requestConfig: ProviderRequestConfig<T>,
+    adapterConfig: AdapterConfig<T['CustomSettings']>,
+  ): Promise<TimestampedProviderResult<T>[]> {
+    try {
+      const requesterResult = await this.requester.request<T['Provider']['ResponseBody']>(
+        requestConfig.params.map((p) => JSON.stringify(p)).join('|'),
```

Comment:
Would there be any problems if this key gets extremely large especially for large batch requests?